src/main/java/frc/robot/subsystems/DrivetrainSubsystem.java

```diff
@@ -17,280 +15,212 @@
+				FRONT_RIGHT_MODULE_STEER_MOTOR,
+				FRONT_RIGHT_MODULE_STEER_ENCODER,
+				FRONT_RIGHT_MODULE_STEER_OFFSET);
+
+		m_backLeftModule = Mk4SwerveModuleHelper.createFalcon500(
+				tab.getLayout("Back Left Module", BuiltInLayouts.kList)
+						.withSize(2, 4)
+						.withPosition(4, 0),
+				Mk4SwerveModuleHelper.GearRatio.L2,
+				BACK_LEFT_MODULE_DRIVE_MOTOR,
+				BACK_LEFT_MODULE_STEER_MOTOR,
+				BACK_LEFT_MODULE_STEER_ENCODER,
+				BACK_LEFT_MODULE_STEER_OFFSET);
+
+		m_backRightModule = Mk4SwerveModuleHelper.createFalcon500(
+				tab.getLayout("Back Right Module", BuiltInLayouts.kList)
+						.withSize(2, 4)
+						.withPosition(6, 0),
+				Mk4SwerveModuleHelper.GearRatio.L2,
+				BACK_RIGHT_MODULE_DRIVE_MOTOR,
+				BACK_RIGHT_MODULE_STEER_MOTOR,
+				BACK_RIGHT_MODULE_STEER_ENCODER,
+				BACK_RIGHT_MODULE_STEER_OFFSET);
+
+		odometry = new SwerveDriveOdometry(m_kinematics, GyroSubsystem.getBestRotation2d());
+	}
+
+	/**
+	 * Sets the gyroscope angle to zero. This can be used to set the direction the
+	 * robot is currently facing to the
+	 * 'forwards' direction.
+	 */
+	public void resetFieldOriented() {
+		calculatedPose = new Pose2d(new Translation2d(0, 0), new Rotation2d(0));
+		odometry.resetPosition(calculatedPose, GyroSubsystem.getBestRotation2d());
+	}
+
+	public Pose2d getPose() {
+		return odometry.getPoseMeters();
+	}
+
+	public void drive(ChassisSpeeds chassisSpeeds) {
+		m_chassisSpeeds = chassisSpeeds;
+	}
+
+	@Override
+	public void periodic() {
+		SwerveModuleState[] states = m_kinematics.toSwerveModuleStates(m_chassisSpeeds);
+		SwerveDriveKinematics.desaturateWheelSpeeds(states, MAX_VELOCITY_METERS_PER_SECOND);
+
+		m_frontLeftModule.set(states[0].speedMetersPerSecond / MAX_VELOCITY_METERS_PER_SECOND * MAX_VOLTAGE,
+				states[0].angle.getRadians());
+		m_frontRightModule.set(states[1].speedMetersPerSecond / MAX_VELOCITY_METERS_PER_SECOND * MAX_VOLTAGE,
+				states[1].angle.getRadians());
+		m_backLeftModule.set(states[2].speedMetersPerSecond / MAX_VELOCITY_METERS_PER_SECOND * MAX_VOLTAGE,
+				states[2].angle.getRadians());
+		m_backRightModule.set(states[3].speedMetersPerSecond / MAX_VELOCITY_METERS_PER_SECOND * MAX_VOLTAGE,
+				states[3].angle.getRadians());
+
+		calculatedPose = odometry.update(GyroSubsystem.getBestRotation2d(), states);
+		//SmartDashboard.putNumber("Estimated X", calculatedPose.getX());
+		//SmartDashboard.putNumber("Estimated Y", calculatedPose.getY());
+		//SmartDashboard.putNumber("\"Estimated\" Rotation", calculatedPose.getRotation().getDegrees());
+		estimatedX.setDouble(calculatedPose.getX());
+		estimatedY.setDouble(calculatedPose.getY());
+		rotation.setDouble(calculatedPose.getRotation().getDegrees());
+
+		//SmartDashboard.putNumber("NavX X", gyroPose.getX());
+		//SmartDashboard.putNumber("NavX Y", gyroPose.getY());
+
+		// Not odometry
+		/*pigeon.getAccumGyro(xyz);
+		SmartDashboard.putNumber("Pigeon X", xyz[0]);
+		SmartDashboard.putNumber("Pigeon Y", xyz[1]);
```

Comment:
Make sure the branch you're requesting to pull is up to date with the latest master changes. These changes are removing heading control.

Reply:
My B I pointed this comment to the wrong line.